kernel/src/sched.rs

```diff
@@ -900,4 +735,247 @@ impl Kernel {
 
         (return_reason, time_executed_us)
     }
+
+    /// Method to invoke a system call on a particular process.
+    /// Applies the kernel system call filtering policy (if any).
+    /// Handles `Yield` and `Exit`, dispatches `Memop` to `memop::memop`,
+    /// and dispatches peripheral driver system calls to peripheral
+    /// driver capsules through the platforms `with_driver` method.
+    #[inline]
+    unsafe fn handle_syscall<P: Platform>(
+        &self,
+        platform: &P,
+        process: &dyn process::ProcessType,
+        syscall: Syscall,
+    ) {
+        // Hook for process debugging.
+        process.debug_syscall_called(syscall);
+        // Enforce platform-specific syscall filtering here.
+        //
+        // Before continuing to handle non-yield syscalls
+        // the kernel first checks if the platform wants to
+        // block that syscall for the process, and if it
+        // does, sets a return value which is returned to
+        // the calling process.
+        //
+        // Filtering a syscall (i.e. blocking the syscall
+        // from running) does not cause the process to loose
+        // its timeslice. The error will be returned
+        // immediately (assuming the process has not already
+        // exhausted its timeslice) allowing the process to
+        // decide how to handle the error.
+        match syscall {
+            Syscall::Yield {
+                which: _,
+                address: _,
+            } => {} // Do nothing
```

Comment:
Should Memop and Exit be filterable?

Reply:
It seems believable that some particularly cautious kernel would want to filter memop calls (to enforce that an app can never malloc, maybe, or to prevent it from accessing debug information about memory addresses).

It seems we should never filter exit